### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/dovecot-documentation/commit/cbfa8b8681931dbfdf6399e00a702fdd8ea94bf3#commitcomment-62569954

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/dovecot-documentation/commit/885f3e058afae8ef5919b1ae2f75905737b6531d

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: source/admin_manual/error_codes.rst [hunk 1]
```diff
@@ -57,11 +57,11 @@ Code  Label          Description
                      this error occurs, would possibly indicate an issue with
                      the sync (i.e. duplicate message). In this case, the
                      migration should be marked as failed and the user flagged
-                     for further investigation why the sync was unsuccesful.
+                     for further investigation why the sync was unsuccessful.
 
 74    EX_IOERR       *Input/Output Error*.
 
-75    EX_TEMPFAILE   *Temporary failure*. A temporary error that may be
+75    EX_TEMPFAIL   *Temporary failure*. A temporary error that may be
```
Comment (jsoref) [inline source/admin_manual/error_codes.rst:64]: Oops, need a space for the column alignment.

File: source/configuration_manual/authentication/authentication_mechanisms.rst [hunk 2]
```diff
@@ -84,7 +84,7 @@ Dovecot supports the following non-plaintext mechanisms:
 +------------------------------------------------------------------------------+--------------------------------------------------------------------------+--------------------------+
 | EXTERNAL                                                                     | EXTERNAL SASL mechanism.                                                 |                          |
 +------------------------------------------------------------------------------+--------------------------------------------------------------------------+--------------------------+
-| OAUTBEARER                                                                   | OAuth2 bearer authentication https://tools.ietf.org/html/rfc7628.        | .. versionadded:: 2.2.29 |
+| OAUTHBEARER                                                                   | OAuth2 bearer authentication https://tools.ietf.org/html/rfc7628.        | .. versionadded:: 2.2.29 |
```
Comment (jsoref) [inline source/configuration_manual/authentication/authentication_mechanisms.rst:87]: I thought I fixed this table.

File: source/configuration_manual/dovemon.rst [hunk 3]
```diff
@@ -60,7 +60,7 @@ timeout period for the next round of polling. This is repeated until the max
 retries have been performed. In the default case where where 3 retries are
 performed with timeout set to 3s, dovemon waits for 3s, 6s, and 9s for first,
 second, and third polling results respectively. However, this does not affect
-the interval between pollings i.e. start time of each polling is always at 10s
+the interval between polls i.e. start time of each polling is always at 10s
```
Comment (jsoref) [inline source/configuration_manual/dovemon.rst:63]: Note: this is a significant change in spelling.

File: source/configuration_manual/sieve/extensions/enotify.rst [hunk 4]
```diff
@@ -18,8 +18,8 @@ Configuration
 Examples
 ========
 
-Send notifications with different importances
----------------------------------------------
+Send notifications with different importance levels
```
Comment (jsoref) [inline source/configuration_manual/sieve/extensions/enotify.rst:21]: This change is to make dictionaries (and me) happy.

File: source/configuration_manual/sieve/plugins/extprograms.rst [hunk 5]
```diff
@@ -265,15 +265,15 @@ redirected in the Sieve script shown above.
 
    # DB STRUCTURE
    #CREATE TABLE `sieve_count` (
-   #  `from_addres` varchar(254) NOT NULL,
+   #  `from_address` varchar(254) NOT NULL,
```
Comment (jsoref) [inline source/configuration_manual/sieve/plugins/extprograms.rst:268]: This is a schema change, but I think it's to an example, so this should be ok.

File: source/developer_manual/design/arrays.rst [hunk 6]
```diff
@@ -219,7 +219,7 @@ reallocate the array elsewhere in memory.
    Return a non-const pointer to the specified index.
    Assert-crashes if the index doesn't already exist.
    See also ``array_idx_get_space()``.
-``array_front_modifiabe(array)``
+``array_front_modifiable(array)``
```
Comment (jsoref) [inline source/developer_manual/design/arrays.rst:222]: This appears to be a correct correction -- the only google hits are for this dovecot documentation.

File: source/developer_manual/development_tips/debugging_tips.rst [hunk 7]
```diff
@@ -252,7 +252,7 @@ being expanded into:
   union {
     struct array arr;
     struct dict_connection_cmd *const *v;
-    struct dict_connection_cmd **v_modifiaable;
+    struct dict_connection_cmd **v_modifiable;
```
Comment (jsoref) [inline source/developer_manual/development_tips/debugging_tips.rst:255]: probably ok?

File: source/installation_guide/dovecot_pro_releases/dovecot-2.3.10.1-rev2/amzn/2/index.rst [hunk 8]
```diff
@@ -52,8 +52,8 @@ OX Dovecot Pro IMAP Proxy Authentication plug-in
 .. literalinclude:: dovecot-2.3.10.1-rev2-imap-proxyauth-plugin.repo
      :language: none
 
-OX Dovecot Pro Pigeonhole Sieve Zimbra compability plug-in
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+OX Dovecot Pro Pigeonhole Sieve Zimbra compatibility plug-in
```
Comment (jsoref) [inline source/installation_guide/dovecot_pro_releases/dovecot-2.3.10.1-rev2/amzn/2/index.rst:55]: I can't tell if this is a real typo from production. The internet only points back to these docs.

If the typo is still present in production, someone should file a bug to the vendor.

File: source/installation_guide/upgrading/from-2.3-to-2.3.x.rst [hunk 9]
```diff
@@ -83,7 +83,7 @@ Upgrading Dovecot v2.3.x to v2.3.15
 ===================================
 
  * :dovecot_core:ref:`ssl_min_protocol` default changed to TLSv1.2, as older TLS versions are deprecated (see `RFC 8996 <https://datatracker.ietf.org/doc/html/rfc8996>`_). Change it to TLSv1 or TLSv1.1 if you need to support older, deprecated protocols.
- * The 'SNIPPET' and 'PREVIEW (w/explicit algorithm selection)' IMAP commands have been deprecated. The new RFC 8970 compliant PREVIEW command should be exclusoively used in the future.
+ * The 'SNIPPET' and 'PREVIEW (w/explicit algorithm selection)' IMAP commands have been deprecated. The new RFC 8970 compliant PREVIEW command should be exclusively used in the future.
```
Comment (jsoref) [inline source/installation_guide/upgrading/from-2.3-to-2.3.x.rst]: This is not the same as the past couple hundred changes

File: source/settings/plugin/apparmor-plugin.rst [hunk 10]
```diff
@@ -26,5 +26,5 @@ Settings
 
      plugin {
        apparmor_hat = hat_name
-       apparamor_hat2 = another_hat
+       apparmor_hat2 = another_hat
```
Comment (jsoref) [inline source/settings/plugin/apparmor-plugin.rst]: interesting

**Review by jsoref — COMMENTED** [review]
Obviously this might want to be squished/split/etc. happy to do whatever.

I'm going to delay fixing the whitespace until I get feedback.

File: source/installation_guide/dovecot_pro_releases/dovecot-2.3.10.1-rev2/centos/6/dovecot-2.3.10.1-rev2-pigeonhole-sieve-zimbra-compat-plugin.repo [hunk 11]
```diff
@@ -1,7 +1,7 @@
 # pigeonhole-sieve-zimbra-compat-plugin component requires separate OX License
 # Please contact to support@open-xchange.com to get access
 [dovecot-2.3.10.1-rev2-pigeonhole-sieve-zimbra-compat-plugin]
-name=CentOS 6 - OX Dovecot Pro Pigeonhole Sieve Zimbra compability plug-in
+name=CentOS 6 - OX Dovecot Pro Pigeonhole Sieve Zimbra compatibility plug-in
```
Comment (cmouse) [inline source/installation_guide/dovecot_pro_releases/dovecot-2.3.10.1-rev2/centos/6/dovecot-2.3.10.1-rev2-pigeonhole-sieve-zimbra-compat-plugin.repo:4]: this content is autogenerated, i'll have to get the template fixed.

**Review by cmouse — CHANGES_REQUESTED** [review]
Thank you for your pull request! It looks really good, and I am happy to merge this with these separate commits once the nits have been addressed.

**Review by cmouse — APPROVED** [review]

**Pull request merged**